dotcom-rendering/src/web/components/Liveness.importable.tsx

```diff
@@ -228,16 +233,20 @@ export const Liveness = ({
 				handleVisibilityChange,
 			);
 		};
-	}, [numHiddenBlocks]);
+	}, [numHiddenBlocks, onFirstPage]);
 
 	const handleToastClick = () => {
-		setShowToast(false);
-		document.getElementById('maincontent')?.scrollIntoView({
-			behavior: 'smooth',
-		});
-		window.location.href = '#maincontent';
-		revealPendingBlocks();
-		setNumHiddenBlocks(0);
+		if (onFirstPage) {
+			setShowToast(false);
+			document.getElementById('maincontent')?.scrollIntoView({
+				behavior: 'smooth',
+			});
+			window.location.href = '#maincontent';
+			revealPendingBlocks();
+			setNumHiddenBlocks(0);
+		} else {
+			window.location.href = `${webURL}#maincontent`;
```

Comment:
I'm trying to think of the flow:
- if not on first page and we have new blocks, toast will show
- if user clicks on toast they are redirected to first page
- on redirection we request paginated blocks for page 1
--> new paginated blocks for page 1 include latest posts?

Reply:
It's more:

- if not on first page and we have new blocks, toast will show
- if user clicks on toast they are redirected to first page

At that point the page already includes the latest blocks so we don't need to do anything. We then start polling as normal. It's essentially as though they just opened the blog for the first time